### ISSUE TITLE
Treat main blueprint as special case / replace with addon hook

### ISSUE THREAD

**Issue by slindberg** [ember-cli/ember-cli#3831]
TL;DR

The main blueprint is often used by addons simply to [add bower dependencies to `bower.json`](https://github.com/emberjs/data/blob/v1.0.0-beta.16.1/lib/ember-addon/blueprints/ember-data/index.js#L12), without generating any app code. This is at odds with the very concept of 'blueprints' and 'generation', and therefore should be served by a dedicated addon hook. I propose `postInstall`. 

-------------------

After updating to v0.2.3 I started getting this error after installing addons:

```
The `ember generate` command requires an entity name to be specified. For more details, use `ember help`.
```

This is happening because of the recently added `addon-import` blueprint that runs after _every_ addon blueprint. This makes sense for the majority of blueprints, but for the main blueprint it doesn't. The main blueprint usually doesn't generate anything, and the concept of an 'entity' doesn't even make sense for it. For some reason, the first arg (after the addon name) of the `install` command [is used as the entity name](https://github.com/ember-cli/ember-cli/blob/v0.2.3/lib/tasks/generate-from-blueprint.js#L36), which is almost always omitted, resulting in the above error.

This has lead to [some](https://github.com/emberjs/data/blob/v1.0.0-beta.16.1/lib/ember-addon/blueprints/ember-data/index.js#L4-L8) [silliness](https://github.com/lytics/ember-data.model-fragments/blob/master/ember-addon/blueprints/ember-data-model-fragments/index.js#L4-L6) in order suppress the error on addon installs when the main blueprint is run. This workaround no longer helps, since the `addon-import` blueprint inherits options (i.e. entity name which is `undefined`) from the main blueprint options, and now no amount of silliness can prevent it from happening.

IMO, this speaks to the main blueprint having reached the end of it's life – it's gotten us this far, but there really should be a specific pathway/hook for adding bower components and/or just running arbitrary code on addon install without contorting blueprints to suit the purpose.

**Comment by trabus** [comment]
Fortunately the error you're seeing is just a silent error, on the `addon-import` blueprint (which shouldn't be running in those instances, I'm looking into fixing that now) which doesn't impede the installation of the other blueprint. Just some annoying noise in this case.

The `generate-from-blueprint` task seems to be taking on additional responsibilities that were not originally intended to be handled by it specifically (hence the assignment of entity, even though there is only one argument in the args array). I would like to look into how this could be extracted into its own task possibly, or at least extract the core functionality used by both and extend it.

In the meantime, I'll implement some silliness to quiet the silent error you are seeing. We definitely need to fix the root issue, but it's a bit more than I'm familiar with regarding the why of how it was implemented.

@jakecraige  @jgwhite  any insights you can provide here would be great.

**Comment by slindberg** [comment]
Thanks for looking into preventing that @trabus. Silent or not, it's displays like any other error on the console, and definitely made me think that the entire blueprint failed.

**Comment by xcambar** [comment]
Although this has landed in `ember-cli 0.2.4`, this occurred to me with `1.13.8` with [ember-material-design](https://github.com/mike-north/ember-material-lite) in a fresh app, as described in https://github.com/mike-north/ember-material-lite/issues/29.

In that particular case, running `ember generate` was necessary to complete the install (Bower deps).

**Comment by trabus** [comment]
I'll take a look at this this afternoon.

**Comment by trabus** [comment]
@xcambar This is a actually because the blueprint has not overridden normalizeEntityName. It's a long standing quirk that I completely forgot about. I'll be making a PR on ember-material-lite shortly.

**Comment by xcambar** [comment]
Very cool.
